mono/metadata/icall.c

```diff
@@ -5888,6 +5892,28 @@ ves_icall_System_Environment_get_NewLine (void)
 #endif
 }
 
+ICALL_EXPORT MonoBoolean
+ves_icall_System_Environment_GetIs64BitOperatingSystem (void)
+{
+#if SIZEOF_VOID_P == 8
+	return TRUE;
+#else
+#ifdef HOST_WIN32
+	MonoBoolean isWow64Process = FALSE;
+	if (IsWow64Process (GetCurrentProcess (), &isWow64Process) {
```

Comment:
I think this is wrong. `MonoBoolean` is a `mono_byte`, which in turn is a `uint8_t`, while `IsWow64Process` parameter is a `BOOL`, aka `int`.